sycl/test-e2e/KernelCompiler/kernel_compiler.cpp

```diff
@@ -0,0 +1,165 @@
+//==- kernel_compiler.cpp --- kernel_compiler extension tests  -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: ocloc
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// the new spec for the kernel_compiler opens the door to supporting several
+// different source languages. But, initially, OpenCL Kernels are the only ones
+// supported. This test is limited to that (thus the cm-compiler requirement)
+// but in the future it may need to broken out into other tests.
+
+#include <sycl/sycl.hpp>
+
+auto constexpr CLSource = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0);
+  out[i] = in[i]*2 + 100;
+}
+__kernel void her_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0);
+  out[i] = in[i]*5 + 1000;
+}
+)===";
+
+auto constexpr BadCLSource = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0) +  no semi-colon!!
+  out[i] = in[i]*2 + 100;
+}
+)===";
+/*
+Compile Log:
+1:3:34: error: use of undeclared identifier 'no'
+  size_t i = get_global_id(0) +  no semi-colon!!
+                                 ^
+1:3:36: error: expected ';' at end of declaration
+  size_t i = get_global_id(0) +  no semi-colon!!
+                                   ^
+                                   ;
+
+Build failed with error code: -11
+
+=============
+
+*/
+
+void testSyclKernel(sycl::queue &Q, sycl::kernel Kernel, int multiplier,
+                    int added) {
+  constexpr int N = 4;
+  int InputArray[N] = {0, 1, 2, 3};
+  int OutputArray[N] = {};
```

Comment:
```suggestion
  cl_int InputArray[N] = {0, 1, 2, 3};
  cl_int OutputArray[N] = {};
```

I recently expanded the extension specification to list the rules for legal parameter types, and I realized that we should really be using the OpenCL types like `cl_int` on the host.  Check the types of your other kernel arguments too.  See 41945cd5d4105b0637860238c6f8ca2b8789a880 for the details I added to the spec.

Reply:
Hi @cperkinsintel,

I saw that you requested re-review of this PR.  Please see this comment I made above yesterday.

Reply:
Sorry about that @gmlueck , I didn't mean to ping you last night.  I made the change now, using `sycl::cl_int`  (etc) instead of `int` and friends in both the kernel_compiler.cpp test and the ocl_capabilities.cpp one.

I think I saw in the recent spec update that the example has `#include <OpenCL/opencl.h>` to get the OpenCL types, but that shouldn't be necessary, correct? They are exposed in the sycl namespace.  Let me know if I'm confusing anything.

Reply:
Actually, the "cl_" types are not supposed to be exposed in the `sycl` namespace.  If our headers are doing that, it is a bug.

The SYCL [OpenCL backend interop](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_data_types_2) spec does say these types are available in the `sycl::opencl` namespace, but that's only if the implementation supports the OpenCL backend.  (DPC++ does, but that's not true for all SYCL implementations.)  It's safest to include `<OpenCL/opencl.h>` and then use the "cl_" types in the root namespace.

Reply:
Thanks. Updated

Reply:
Don't you also need `#include <OpenCL/opencl.h>`?